src/main/java/org/gridsuite/modification/server/repositories/NetworkModificationRepository.java

```diff
@@ -0,0 +1,124 @@
+
+/**
+ * @author Slimane Amar <slimane.amar at rte-france.com>
+ */
+@Repository
+public class NetworkModificationRepository {
+    private ModificationGroupRepository modificationGroupRepository;
+
+    private ModificationRepository modificationRepository;
+
+    public NetworkModificationRepository(ModificationGroupRepository modificationGroupRepository, ModificationRepository modificationRepository) {
+        this.modificationGroupRepository = modificationGroupRepository;
+        this.modificationRepository = modificationRepository;
+    }
+
+    @Transactional // To have the 2 delete in the same transaction (atomic)
+    public void deleteAll() {
+        modificationRepository.deleteAll();
+        modificationGroupRepository.deleteAll();
+    }
+
+    @Transactional // To have the 2 create in the same transaction (atomic)
+    public ElementaryModificationEntity createElementaryModification(UUID groupUuid, String equipmentId, Set<String> substationId,
+                                                                     String attributeName, Object attributeValue) {
+        ElementaryModificationEntity elementaryModificationEntity = new ElementaryModificationEntity(equipmentId, substationId, createAttributeEntity(attributeName, attributeValue));
+        ModificationGroupEntity modificationGroupEntity = this.modificationGroupRepository.findById(groupUuid).orElse(createModificationGroup(groupUuid));
+        elementaryModificationEntity.setGroup(modificationGroupEntity);
+        this.modificationRepository.save(elementaryModificationEntity);
+        return elementaryModificationEntity;
+    }
+
+    private ModificationGroupEntity createModificationGroup(UUID groupUuid) {
+        return modificationGroupRepository.save(new ModificationGroupEntity(groupUuid));
+    }
+
+    public List<UUID> getModificationGroupsUuids() {
+        return this.modificationGroupRepository.findAll().stream()
+                .map(ModificationGroupEntity::getUuid)
+                .collect(Collectors.toList());
+    }
+
+    public List<ModificationInfos> getModifications(UUID groupUuid) {
+        ModificationGroupEntity group = getModificationGroup(groupUuid);
+        return this.modificationRepository.findAll(group.getUuid())
+                .stream()
+                .map(ModificationEntity::toModificationInfos)
+                .collect(Collectors.toList());
+    }
+
+    @Transactional(readOnly = true)
+    // Due to the lazy loading done outside transaction (demarcation done in findAllByGroupUuid method)
+    public List<ElementaryModificationInfos> getElementaryModifications(UUID groupUuid) {
+        ModificationGroupEntity group = getModificationGroup(groupUuid);
+        return this.modificationRepository.findAllByGroupUuid(group.getUuid())
+                .stream()
+                .filter(m -> m.getType().equals(ModificationType.ELEMENTARY.name()))
```

Comment:
Do this filter in the sql ?

Reply:
I think we will never need to get only elementary modifications but rather all modifications in a group

Reply:
Done